### ISSUE TITLE
Question about SMILES preprocessing in KPGT pretraining (ChEMBL normalization)

### ISSUE THREAD

**Issue by Serious-Huang** [lihan97/KPGT#17]
Hi,

I have a question regarding how SMILES strings were preprocessed during `KPGT` pretraining. Since `KPGT` was pretrained on the `ChEMBL` database, I’m wondering what normalization or standardization steps were applied to the SMILES before training.

**For example:**
- Were charged molecules neutralized or standardized in a specific way?

As different charged forms or SMILES strings of the same molecule can result in different molecular representations, the preprocessing pipeline may significantly affect the learned embeddings.

Does this mean that, for downstream applications, users should apply the same preprocessing pipeline to generate reasonable molecular representations? If so, is there any documentation or code that specifies the exact preprocessing steps used? I may have missed this in the paper or code, but any clarification would be greatly appreciated.

Thanks in advance for your clarification!

**Comment by lihan97** [comment]
For the preprocessing steps applied to SMILES strings from the ChEMBL database during KPGT pretraining, please refer to our preprocessing script here: https://github.com/lihan97/KPGT/blob/main/scripts/preprocess_pretrain_dataset.py. And for preprocessing scripts for finetuning, please refer to https://github.com/lihan97/KPGT/blob/main/scripts/preprocess_downstream_dataset.py.

**Comment by Serious-Huang** [comment]
> For the preprocessing steps applied to SMILES strings from the ChEMBL database during KPGT pretraining, please refer to our preprocessing script here: https://github.com/lihan97/KPGT/blob/main/scripts/preprocess_pretrain_dataset.py. And for preprocessing scripts for finetuning, please refer to https://github.com/lihan97/KPGT/blob/main/scripts/preprocess_downstream_dataset.py.

Thank you for the prompt response. Then KPGT uses the SMILES directly from the ChEMBL database, which are RDKit-generated canonical SMILES. That answers my question. Thanks!
